### PR TITLE
2.4 PXB-2297 5.7.32 Merge

### PR DESCRIPTION
**Pull request by altmannmarcelo**
https://jira.percona.com/browse/PXB-2297

### CODE REVIEW

**Comment by altmannmarcelo** [comment]
https://pxb.cd.percona.com/job/percona-xtrabackup-2.4-param-medium/125/

File: storage/innobase/fsp/fsp0fsp.cc [hunk 1]
```diff
@@ -1232,7 +1232,14 @@ fsp_header_decode_encryption_info(
...
 			    ENCRYPTION_MAGIC_SIZE) == 0) {
-		return (true);
+		/* We ignore report error for recovery,
+		since the encryption info maybe hasn't writen
+		into datafile when the table is newly created. */
+		if (!recv_recovery_is_on()) {
+			return(false);
+		} else {
+			return(true);
+		}
```
Comment (satya-bodapati) [inline storage/innobase/fsp/fsp0fsp.cc]: return(recv_recovery_is_on() ? true : false);

**Review by satya-bodapati — APPROVED** [review]
minor suggestion

**Comment by altmannmarcelo** [comment]
New Jenkins run after rebase - https://pxb.cd.percona.com/job/percona-xtrabackup-2.4-param-medium/128/

File: storage/innobase/fsp/fsp0fsp.cc [hunk 2]
```diff
@@ -1232,10 +1232,12 @@ fsp_header_decode_encryption_info(
 		version = Encryption::ENCRYPTION_VERSION_3;
 	} else if (memcmp(ptr, ENCRYPTION_KEY_MAGIC_EMPTY,
 			    ENCRYPTION_MAGIC_SIZE) == 0) {
-		return (true);
+		/* We ignore report error for recovery,
+			into datafile when the table is newly created. */
```
Comment (satya-bodapati) [inline storage/innobase/fsp/fsp0fsp.cc]: Align the comment please

**Review by satya-bodapati — APPROVED** [review]
LGTM! minor comment

**Pull request merged**